packages/fluentui/react-northstar/src/components/Tree/Tree.tsx

```diff
@@ -245,186 +254,188 @@ class Tree extends AutoControlledComponent<WithAsProp<TreeProps>, TreeState> {
               if (item.items) {
                 selectItems(item.items);
               } else if (selectble) {
-                selectedItemIds.push(item.id);
+                nextSelectedItemIds.push(item.id);
               }
             }
           });
         };
         selectItems(items);
       }
-
-      this.setSelectedItemIds(e, selectedItemIds);
+      setSelectedItemIds(e, [...nextSelectedItemIds]);
       return;
     }
 
     // push/remove single tree item into selection array
     if (selectedItemIds.indexOf(id) === -1) {
-      selectedItemIds.push(id);
+      nextSelectedItemIds = [...selectedItemIds, id];
     } else {
-      selectedItemIds = selectedItemIds.filter(itemID => itemID !== id);
+      nextSelectedItemIds = nextSelectedItemIds.filter(itemID => itemID !== id);
     }
 
-    this.setSelectedItemIds(e, selectedItemIds);
+    setSelectedItemIds(e, nextSelectedItemIds);
   };
 
-  onTitleClick = (e: React.SyntheticEvent, treeItemProps: TreeItemProps, executeSelection: boolean = false) => {
+  const onTitleClick = (e: React.SyntheticEvent, treeItemProps: TreeItemProps, executeSelection: boolean = false) => {
```

Comment:
Still unstable

Reply:
Setting it in the useCallback it was breaking

Reply:
Solved